Week_12/Ejercicio_1.py

```diff
@@ -0,0 +1,29 @@
+class BankAccount:
+    balance = 0
+
+    def add_money(self, amount):
+        self.balance += amount
+        print(f"Transaccion satisfactoria, el saldo actual de su cuenta es {self.balance}")
+
+    def subtract_money(self, amount):
+        if amount > self.balance:
+            print(f"Fondos insuficientes, usted dispone de {self.balance}")
+        else:
+            self.balance -= amount
+
+class SavingsAccount(BankAccount):
+    min_balance = 8000
+
+    def check_balance(account):
```

Comment:
Logic is good here, except for the `input` part.
Classes should be reusable and have no direct implementation.
Instead of asking the user for the quantity from inside the method, make the quantity a parameter, just like in the `subtract_money` method of the `BankAccount`, and ask for it outside.

BTW, this method should also be called `subtract_money` to **overwrite** the original. Otherwise, developers would be able to use _the old one_ and bypass the `minimum_balance` attribute.

Reply:
Changes made as per recommendations:
![image](https://github.com/JorgeA/DUAD/assets/4186222/ada6694b-3ae1-45db-9dc0-c3ce7c7066ba)

Reply:
Followed the recommendations:
![image](https://github.com/JorgeA/DUAD/assets/4186222/7687fee7-d840-44c3-a09a-f52a871a7fed)